ipywidgets/static/widgets/js/widget_box.js

```diff
@@ -16,8 +16,74 @@ define([
         }, widget.WidgetModel.serializers)
     });
 
+    var ProxyModel = widget.WidgetModel.extend({}, {
+        serializers: _.extend({
+            child: {deserialize: widget.unpack_models},
+        }, widget.WidgetModel.serializers),
+    });
+
+    var ProxyView = widget.WidgetView.extend({
+        initialize: function() {
+            // Public constructor
+            ProxyView.__super__.initialize.apply(this, arguments);
+            this.$box = this.$el;
+            this.child_promise = Promise.resolve();
+        },
+
+        render: function() {
+            var that = this;
+            var child_view = this.set_child(this.model.get("child"))
+            this.listenTo(this.model, "change:child", function(model, value) {
+                this.set_child(value);
+            });
+            return child_view;
+        },
+
+        remove: function() {
+            ProxyView.__super__.remove.apply(this, arguments);
+            this.child_promise.then(this.child.remove);
+        },
+
+        set_child: function(value) {
+            if (this.child) {
+                this.child.remove();
+            }
+            if (value) {
+                var that = this;
+                this.child_promise = this.child_promise.then(function() {
+                    return that.create_child_view(value).then(function(view) {
+                        if (that.$box.length === 0) {
+                            console.error("Widget place holder does not exist");
+                            return;
+                        }
+                        that.$box.empty().append(view.el);
+                        // Trigger the displayed event of the child view.
+                        that.displayed.then(function() {
+                            view.trigger('displayed');
+                        });
+                        that.child = view;
+                        that.trigger("child:created");
+                    }).catch(utils.reject("Couldn't add child view to proxy", true));
+                });
+            }
+            return this.child_promise;
+        },
+    });
+
+    var PlaceView = ProxyView.extend({
+        initialize: function() {
+            PlaceView.__super__.initialize.apply(this, arguments);
+            this.update_selector(this.model, this.model.get("selector"));
+            this.listenTo(this.model, "change:selector", this.update_selector);
+        },
+
+        update_selector: function(model, selector) {
+            this.$box = selector ? $(selector) : this.$el;
```

Comment:
Don't you need to reset the child so it gets moved to the correct selector when the selector is changed?